tests/PHPUnit/Integration/Tracker/VisitTest.php

```diff
@@ -102,6 +102,45 @@ public function testIsVisitorIpExcluded($excludedIp, $tests)
         }
     }
 
+    public function getExcludeByUrlData()
+    {
+        return array(
+            array(array('http://test.com'), true, array(
+                'http://test.com' => true,
+                'https://test.com' => true,
+                'http://test.com/uri' => true,
+                'http://test.com/?query' => true,
+                'http://xtest.com' => false,
+                'http://x.test.com' => false,
```

Comment:
No problem. Reading issue twice is a very good idea though. Also, it's the beauty of unit test, that we can easily notice how the algorithm works :+1: